kimchi/src/circuits/polynomials/foreign_mul/mod.rs

```diff
@@ -0,0 +1,104 @@
+//! Foreign field multiplication gate
+
+/// <https://hackmd.io/XZUHHGpDQsSOs0dUGugB5w>
+///
+/// Globals:
+///     * n: native field modulus
+///     * p: foreign field modulus
+///
+/// Inputs:
+///     * a: left foreign field element operand a \in Fp
+///     * b: right foreign field element operand b \in Fp
+///
+/// Witness:
+///     * q: \in Fp
+///     * r: such that a*b = q*p +r
+///
+/// Foreign field element structure:
+///
+///    Each foreign field element a is decomposed into three 88-bit limbs a0, a1, a2 s.t. a = a0a1a2 in
+///    little-endian byte order (i.e. a = a2*2^{2b} + a1*2^b + a0)
+///      a0 = a0p0a0p1a0p2a0p3a0p4a0p5a0c0a0c1a0c2a0c3a0c4a0c5a0c6a0c7
```

Comment:
I'm not sure that the numbering helps much here. For me, it would be clearer to have
```
L is a 12-bit lookup,
C is a 2-bit crumb.
     <----6----> <------8------>
a0 = L L L L L L C C C C C C C C
a1 = L L L L L L C C C C C C C C
     <--4--> <------------------20----------------->
a2 = L L L L C C C C C C C C C C C C C C C C C C C C
```

At a minimum, the repeated `a0`/`a1`/`a2` prefix adds a bunch of noise for not much benefit.

Reply:
Agree - thanks!